README
NAME
    Text::Truncate - Perl module with simple string truncating routine

REQUIREMENTS
    This module only uses standard modules. It should run on Perl 5.004.

INSTALLATION
    Installation is pretty standard:

      perl Makefile.PL
      make test
      make install

    (On Windows platforms you should use nmake instead.)

    Using Build.PL (if you have Module::Build installed):

      perl Build.PL
      perl Build test
      perl Build install    

SYNOPSIS
      use Text::Truncate;

      my $long_string = "This is a very long string";

      # outputs "This is..."
      print truncstr( $long_string, 10);

      # outputs "This is a-"
      print truncstr( $long_string, 10, "-");

      # outputs "This is a "
      print truncstr( $long_string, 10, "");

      # outputs "This is---"
      $Text::Truncate::DEFAULT_MARKER = "---";
      print truncstr( $long_string, 10);

DESCRIPTION
    This is a simple, no-brainer subroutine to truncate a string and add an
    optional cutoff marker (defaults to ``...'').

    (Yes, this is a really brain-dead sort of thing to make a module out of,
    but then again, I use it so often that it might as well be in a module.)

    The synopsis gives examples of how to use it.

REVISION HISTORY
    The following changes have been made since the previous release:

    1.03  Sun Jun 26 2005
            - README generated from Pod::Readme
            - minor updates to POD
            - uses Test::Pod::Coverage instead
                  (thanks to Richard Clamp for pointing this out)

    See the Changes file for a detailed revision history.

REPOSITORY
 
 https://github.com/ileiva/Text-Truncate

AUTHOR
    Robert Rothenberg <rrwo at cpan.org>
    This is now being mantained by israel leiva <ilv@cpan.org>

LICENSE
    Unrestricted. This module is in the public domain. No copyright is claimed.



Changes
Revision history for Perl extension Text::Truncate.

1.05  2014-02-09
    - ILV has taken over maintenance
    - Added github repo to metadata and pod
    - Reformatted Changes according to CPAN::Changes::Spec
    - Added LICENSE file
    - Specified 'unrestricted' license in pod
    - Added 'use warnings'

1.04  2009-11-02
 	- re-added to CPAN after accidental removal

1.03  2005-06-26
	- README generated from Pod::Readme
	- minor updates to POD
	- uses Test::Pod::Coverage instead
          (thanks to Richard Clamp for pointing this out)

1.02  2005-04-04
	- fixed bug in Pod::Coverage tests
	- minor updates to POD and README

1.01  2005-03-29
	- added Test::Prereq to tests
	- added Test::Pod tests
	- added Pod::Coverage tests
	- minor cleanup of tests
	- uses Test::More
	- added SIGNATURE

1.00  2004-08-03
	- changes to handle undefined values
	- fixed tests
	- added Build.PL and META.yml to distribution
	- changed minimum version to 5.004
	- minor changes to POD

0.01  2002-11-22
	- original version; created by h2xs 1.21 with options
		-X -A -v 0.01 -n Text::Truncate -b 5.5.0



lib/Text/Truncate.pm
package Text::Truncate;

use 5.00400;
use warnings;
use strict;

use Carp;

require Exporter;
use vars qw( $VERSION @ISA @EXPORT @EXPORT_OK $DEFAULT_MARKER );
@ISA = qw( Exporter );

@EXPORT_OK = ( @EXPORT, qw( $DEFAULT_MARKER ) );
@EXPORT = qw(
 truncstr
);
$VERSION = '1.05';

$DEFAULT_MARKER = "...";

sub truncstr
  {
    my $string = shift;
    my $cutoff = shift;
    my $marker = shift;

    $marker = ($DEFAULT_MARKER || ""), unless (defined($marker));

    croak "continued symbol is longer than the cutoff length",
      if (length($marker) > $cutoff);

    if (length($string) > $cutoff) {
      $string = (substr($string, 0, $cutoff-length($marker))||"") . $marker;
    }

    return $string;
  }

1;
__END__

=head1 NAME

Text::Truncate - Perl module with simple string truncating routine

=begin readme

=head1 REQUIREMENTS

This module only uses standard modules. It should run on Perl 5.004.

=head1 INSTALLATION

Installation is pretty standard:

  perl Makefile.PL
  make test
  make install

(On Windows platforms you should use nmake instead.)

Using Build.PL (if you have L<Module::Build> installed):

  perl Build.PL
  perl Build test
  perl Build install    

=end readme

=head1 SYNOPSIS

  use Text::Truncate;

  my $long_string = "This is a very long string";

  # outputs "This is..."
  print truncstr( $long_string, 10);

  # outputs "This is a-"
  print truncstr( $long_string, 10, "-");

  # outputs "This is a "
  print truncstr( $long_string, 10, "");

  # outputs "This is---"
  $Text::Truncate::DEFAULT_MARKER = "---";
  print truncstr( $long_string, 10);

=head1 DESCRIPTION

This is a simple, no-brainer subroutine to truncate a string and add
an optional cutoff marker (defaults to ``...'').

(Yes, this is a really brain-dead sort of thing to make a module out
of, but then again, I use it so often that it might as well be in a
module.)

The synopsis gives examples of how to use it.

=for readme stop

=head2 EXPORT

The following functions are exported:

=over

=item truncstr

  $rstring = truncstr( $istring, $cutoff_length );

  $rstring = truncstr( $istring, $cutoff_length, $continued_symbol );

If the C<$istring> is longer than the C<$cutoff_length>, then the
string will be truncated to C<$cutoff_length> characters, including
the C<$continued_symbol> (which defaults to ``...'' if none is
specified).

The default C<$continued_symbol> can be changed in
C<$Text::Truncate::DEFAULT_MARKER>.

=back

=for readme continue

=begin readme

=head1 REVISION HISTORY

The following changes have been made since the previous release:

=for readme include file="Changes" type="text" start="^1.04" stop="^1.02"

See the F<Changes> file for a detailed revision history.

=end readme

=head1 REPOSITORY
 
 https://github.com/ileiva/Text-Truncate

=head1 AUTHOR

Robert Rothenberg <rrwo at cpan.org>
This is now being mantained by israel leiva <ilv@cpan.org>

=head1 LICENSE

Unrestricted. This module is in the public domain. No copyright is claimed.

=cut
